### ISSUE TITLE
BMP support is unreliable

### ISSUE THREAD

**Issue by ventsin** [cnjinhao/nana#657]
I've been testing parts of the library and I consistently failed to load BMPs correctly, be it from memory or from the filesystem.

```cpp
nana::paint::image img;
if (!img.open(data, size))
    return;
//or
if (!img.open("test.bmp"))
    return;

nana::picture pic;
pic.create(fm);
if (!pic.load(img))
    return;
pic.move(150, 10);
pic.size({ 32, 32 });
```

Both `open` and `load` return `true`, however the result is always a completely gray image!

On that note, I want to ask @cnjinhao and @qPCR4vir what your thoughts are on using [STB Image](https://github.com/nothings/stb) for loading routines instead of supporting formats directly? It should provide robust reading of images and since it's header-only, it would simplify nana building/installing immensely, since it wouldn't rely on `libpng` and `libjpg` directly.

I wouldn't mind working on a PR but I'd like to know your thoughts before working on it.

**Comment by cnjinhao** [comment]
It’s just a bug on support of a specific BMP format. Would you please send me the bmp file to my email?

**Comment by ventsin** [comment]
[test.zip](https://github.com/cnjinhao/nana/files/8742318/test.zip)
I don't know your address, so I uploaded the file here. If you need it via email let me know.

**Comment by cnjinhao** [comment]
Fixed! New fix has been checked into branch `develop-1.8`.
